tests/interfaces/default_qubit_2_integration/test_autograd_default_qubit_2.py

```diff
@@ -155,99 +161,109 @@ def cost(a):
         tape = qml.tape.QuantumScript([qml.RY(a, wires=0)], [qml.expval(qml.PauliZ(0))])
         tape.trainable_params = [0]
         tapes, fn = param_shift(tape)
-        expected = fn(dev.execute(tapes))
+        expected = fn(device.execute(tapes))
 
         assert expected.shape == ()
-        assert np.allclose(res, expected, atol=tol, rtol=0)
-        assert np.allclose(res, -np.sin(a))
+        assert np.allclose(res, expected, atol=atol_for_shots(shots), rtol=0)
+        assert np.allclose(res, -np.sin(a), atol=atol_for_shots(shots))
 
-    def test_jacobian(self, execute_kwargs, tol):
+    def test_jacobian(self, execute_kwargs, shots, device):
         """Test jacobian calculation"""
         a = np.array(0.1, requires_grad=True)
         b = np.array(0.2, requires_grad=True)
 
-        def cost(a, b, device):
+        def cost(a, b):
             ops = [qml.RY(a, wires=0), qml.RX(b, wires=1), qml.CNOT(wires=[0, 1])]
             m = [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))]
-            tape = qml.tape.QuantumScript(ops, m)
+            tape = qml.tape.QuantumScript(ops, m, shots=shots)
             return autograd.numpy.hstack(execute([tape], device, **execute_kwargs)[0])
 
-        dev = DefaultQubit2()
-
-        res = cost(a, b, device=dev)
+        res = cost(a, b)
         expected = [np.cos(a), -np.cos(a) * np.sin(b)]
-        assert np.allclose(res, expected, atol=tol, rtol=0)
+        assert np.allclose(res, expected, atol=atol_for_shots(shots), rtol=0)
 
-        res = qml.jacobian(cost)(a, b, device=dev)
+        res = qml.jacobian(cost)(a, b)
         assert isinstance(res, tuple) and len(res) == 2
         assert res[0].shape == (2,)
         assert res[1].shape == (2,)
 
         expected = ([-np.sin(a), np.sin(a) * np.sin(b)], [0, -np.cos(a) * np.cos(b)])
-        assert all(np.allclose(_r, _e, atol=tol, rtol=0) for _r, _e in zip(res, expected))
+        for _r, _e in zip(res, expected):
+            assert np.allclose(_r, _e, atol=atol_for_shots(shots))
 
     @pytest.mark.filterwarnings("ignore:Attempted to compute the gradient")
```

Comment:
Is this `filterwarnings` marker still necessary? Same as below.

Reply:
actually, not sure what this warning is about any more.

Reply:
Ah. The first tape has no trainable parameters.  So yes the warning is still there.